analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -147,4 +147,23 @@ public override void ExecutionCompleted()
             ReportIssue(operation, operation.Syntax.ToString());
         }
     }
+
+    private static ProgramState SetConstraintForObjectCreation(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        FirstArgumentAsCollection(objectCreation) switch
```

Comment:
To give it name based on what it gives us, instead of description of how it does that. 
And it doesn't need to be the 1st one when calling stuff out of order.
```suggestion
        CollectionArgument(objectCreation) switch
```